From 80e251f4c30d9f8263c472e07d18a4b0b21ebf4e Mon Sep 17 00:00:00 2001
From: "Bryn M. Reeves" <bmr@redhat.com>
Date: Fri, 14 Feb 2014 17:20:25 +0000
Subject: [PATCH 22/72] Eliminate hard-coded /var/log/sa paths in sar plugin

The Sar plugin classes define a self.sa_path - use it consistently
throughout the plugin when generating strings containing the path.

Suggested by David Kutalek.

Signed-off-by: Bryn M. Reeves <bmr@redhat.com>
---
 sos/plugins/sar.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/sos/plugins/sar.py b/sos/plugins/sar.py
index cf53d5b..470d82f 100644
--- a/sos/plugins/sar.py
+++ b/sos/plugins/sar.py
@@ -32,14 +32,14 @@ class Sar(Plugin,):
         if self.get_option("all_sar"):
             self.sa_size = 0
 
-        self.add_copy_spec_limit("/var/log/sa/sar[0-9]*",
+        self.add_copy_spec_limit("%s/sar[0-9]*" % self.sa_path,
                                  sizelimit = self.sa_size)
-        self.add_copy_spec_limit("/var/log/sa/sa[0-9]*",
+        self.add_copy_spec_limit("%s/sa[0-9]*" % self.sa_path,
                               sizelimit = self.sa_size)
         try:
             dirList = os.listdir(self.sa_path)
         except:
-            self.soslog.warning("sar: could not list /var/log/sa")
+            self.soslog.warning("sar: could not list %s" % self.sa_path)
             return
         # find all the sa file that don't have an existing sar file
         for fname in dirList:
-- 
1.9.3

